### DISCUSSION TITLE
Where are job queued when a job is retried after it failed ?

### DISCUSSION THREAD

**Discussion by Aghamyr** [OptimalBits/bull#1963] [Q&A]
No bug here, just a question as I've not found a clear answer to this in the documentation

So I have jobs put in queue with a delay and 2 retries allowed with a minute backoff, the queue is with only one concurrency , some jobs takes longer to finish so the waiting list get sometime backed up a bit. 

My question is this: when an active job fails, does it goes back to delayed or to waiting ? And does it goes first is that list or last or is it base on the other jobs start time and the failed job delay ?

**Comment by manast** [comment]
A job that fails will go to the failed set. If it is retried it will go to the wait list and if it has an automatic retry with backoff it will go to the delay set.

**Reply by emaxters** [reply]
Will it be added at the end (tail) of the waiting queue or at the start (head)?
We use 
```
"backoff": {
    "type": "fixed",
    "delay": 3600000
  }
  ```

**Reply by manast** [reply]
They will be added to the head, so that they get processed as soon as possible
